support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/SupporterPlusSubcriptionBuilder.scala

```diff
@@ -27,6 +28,55 @@ class SupporterPlusSubcriptionBuilder(
       csrUsername: Option[String],
       salesforceCaseId: Option[String],
   ): Either[PromoError, SubscribeItem] = {
+    if (state.product.fulfilmentOptions == NoFulfilmentOptions) {
+      createStandardSupporterPlus(state, csrUsername, salesforceCaseId)
+    } else {
+      createSupporterPlusWithGuardianWeekly(state, csrUsername, salesforceCaseId)
+    }
+  }
+
+  private def createSupporterPlusWithGuardianWeekly(
+      state: SupporterPlusState,
+      csrUsername: Option[String],
+      salesforceCaseId: Option[String],
+  ) = {
+    val productRatePlanId =
+      validateRatePlan(supporterPlusRatePlan(state.product, environment), state.product.describe)
+
+    if (state.product.amount != getBaseProductPrice(productRatePlanId, state.product.currency)) {
+      throw new BadRequestException(
+        s"The amount passed in (${state.product.amount}) does not match the price of this product.",
+      )
+    }
+
+    val todaysDate = dateGenerator.today
+
+    val subscriptionData = subscribeItemBuilder.buildProductSubscription(
+      productRatePlanId = productRatePlanId,
+      contractEffectiveDate = todaysDate,
+      contractAcceptanceDate = todaysDate,
+      readerType = Direct,
```

Comment:
Just because I still don't really know what `readerType` is - is this always `Direct`?

Reply:
Yes it will always be `Direct` the only other value of this that we care about is `Gift` which is used for gift subs and we don't have those for this product